src/noit_check_lmdb.c

```diff
@@ -67,11 +73,17 @@ static int noit_check_lmdb_add_attribute(xmlNodePtr root, xmlNodePtr attr, noit_
 static int noit_check_lmdb_add_config(xmlNodePtr root, xmlNodePtr config, noit_lmdb_check_data_t *key_data, MDB_val mdb_data) {
   mtevAssert(root != NULL);
   mtevAssert(config != NULL);
-  if ((!mdb_data.mv_data) || (mdb_data. mv_size == 0)) {
+  char *val = NULL;
+  if ((mdb_data.mv_data == NULL) && (mdb_data.mv_size > 0)) {
+    /* If data is null, but we have a size, something is wrong - just skip
+     * this key */
+    mtevL(mtev_error, "noit_check_lmdb_add_config: got null data for key with a size, skipping\n");
     return 0;
   }
-  char *val = (char *)calloc(1, mdb_data.mv_size + 1);
-  memcpy(val, mdb_data.mv_data, mdb_data.mv_size);
+  if ((mdb_data.mv_data != NULL) && (mdb_data.mv_size > 0)) {
+    val = (char *)calloc(1, mdb_data.mv_size + 1);
+    memcpy(val, mdb_data.mv_data, mdb_data.mv_size);
```

Comment:
```suggestion
    memcpy(val, mdb_data.mv_data, mdb_data.mv_size);
    val[mdb_data.mv_size] = 0;
```

Reply:
We don't need to do that. We're using calloc, which automatically memsets the data to zero.

Reply:
kk